autonomy/configurations/base.py

```diff
@@ -207,179 +199,153 @@ def validate_config_data(
-        """Check that overrides are valid.
-
-        - number of overrides is 1
-        - number of overrides == number of agents in spec
-        - number of overrides is 0
-
-        :param overrides: List of overrides
-        :return: True if overrides are valid
-        """
-        valid = []
-        remaining = copy(overrides)
-
-        for component in [
-            CONNECTION,
-            CONTRACT,
-            PROTOCOL,
-            SKILL,
-        ]:
-
-            component_overrides = [f for f in overrides if f["type"] == component]
-            remaining = [f for f in remaining if f not in component_overrides]
-            if any(
-                [
-                    self.number_of_agents == len(component_overrides),
-                    len(component_overrides) == 0,
-                    len(component_overrides) == 1,
-                ]
-            ):
-                valid.append(True)
-
-        if len(remaining) > 0:  # pragma: nocover
-            raise ValueError(f"Override type is misspelled.\n {remaining}")
-
-        if sum(valid) == 4:
-            return True
-
-        raise ValueError("Incorrect number of overrides for count of agents.")
-
     def check_overrides_valid(
         self, overrides: List, env_vars_friendly: bool = False
-    ) -> Dict[ComponentId, Dict[Any, Any]]:
+    ) -> None:
         """Uses the AEA helper libraries to check individual overrides."""
-
-        component_configurations: Dict[ComponentId, Dict] = {}
-        # load the other components.
-
-        for idx, component_configuration_json in enumerate(overrides):
-            component_id, _ = self.process_component_section(
-                idx, component_configuration_json
+        base_validator = validation.ConfigValidator("definitions.json")
+        processed = []
+        for component_configuration_json in overrides:
+            configuration, component_id, override_type = self.process_metadata(
+                configuration=copy(component_configuration_json)
             )
-            if component_id in component_configurations:
-                raise ValueError(
-                    f"Configuration of component {component_id} occurs more than once."
+            if component_id in processed:
+                raise AEAValidationError(
+                    f"Overrides for component {component_id} are defined more than once"
                 )
-            component_configurations[component_id] = component_configuration_json
+            if override_type == OverrideType.MULTIPLE:
+                for idx in range(self.number_of_agents):
+                    try:
+                        _configuration = cast(Dict, configuration[idx])
+                    except KeyError as e:
+                        raise AEAValidationError(
+                            f"Not enough overrides for component {component_id};"
+                            f" Number of agents: {self.number_of_agents}"
+                        ) from e
```

Comment:
can we not simply check `len(configuration) == self.number_of_agents`? Otherwise might still end up specifying more config, e.g. for a 5th agent, plus it will be cleaner as a bonus.

Reply:
```
            if override_type == OverrideType.MULTIPLE:
                enforce(len(configuration) == self.number_of_agents)
            else:
                configuration = (configuration,)
            base_validator.validate_component_configuration(
                component_id=component_id,
                configuration=configuration,
                env_vars_friendly=env_vars_friendly,
            )
            processed.append(component_id)
```

Reply:
If we go with `len(configuration) == self.number_of_agents`, this might lead to an issue where if a user defines an agent index wrong it will raise an error. For example let's say `number_of_agents` is 4 and an user defined indexes as `(0, 1, 2, 3, 5)` the check will still pass and will raise error later